.jenkins/common_job_properties.groovy

```diff
@@ -138,8 +138,11 @@ class common_job_properties {
   // Sets common config for Maven jobs.
   static def setMavenConfig(def context) {
     context.mavenInstallation('Maven 3.3.3')
+    context.mavenOpts('-client')
     context.mavenOpts('-Dorg.slf4j.simpleLogger.showDateTime=true')
     context.mavenOpts('-Dorg.slf4j.simpleLogger.dateTimeFormat=yyyy-MM-dd\\\'T\\\'HH:mm:ss.SSS')
+    context.mavenOpts('-XX:+TieredCompilation')
+    context.mavenOpts('-XX:TieredStopAtLevel=1')
     context.rootPOM('pom.xml')
```

Comment:
By default the JVM on linux runs in server mode and optimizes the JIT for long running processes (it records more statistics about the execution to profile things afterwards). Maven processes are not long running processes so this optimization costs some extra time when you start the JVM.
The idea comes from point 5 here, and if you test it you can see the difference.
https://zeroturnaround.com/rebellabs/your-maven-build-is-slow-speed-it-up/

Note however that the produced code is not different because this changes the options of the JVM for the maven execution not for the java compiler (I compared the generated classes with and without this flags to see if there were differences).